eras/alonzo/impl/src/Cardano/Ledger/Alonzo/UTxO.hs

```diff
@@ -327,5 +328,5 @@ getAlonzoWitsVKeyNeeded ::
   Set.Set (KeyHash 'Witness)
 getAlonzoWitsVKeyNeeded certState utxo txBody =
   getShelleyWitsVKeyNeeded certState utxo txBody
-    `Set.union` (txBody ^. reqSignerHashesTxBodyG)
+    `Set.union` Set.map asWitness (txBody ^. reqSignerHashesTxBodyG)
```

Comment:
this function takes other real `Witness` inputs, so I didn't propagate the `Guard`s any further